### PR TITLE
Support bigquery DDL optional args

### PR DESCRIPTION
**Pull request by m-ahmadkhan**
Listed all possible create/alter/drop commands, and keywords for bigquery. https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language

Will handle the other dialects later in a separate PR.

Also fixes the `AS` issue in https://github.com/sql-formatter-org/sql-formatter/issues/309.

### CODE REVIEW

File: src/languages/bigquery/bigquery.formatter.ts [hunk 1]
```diff
@@ -33,57 +33,116 @@ const reservedCommands = [
...
-  'ALTER SCHEMA SET OPTIONS',
-  'ALTER TABLE SET OPTIONS',
-  'ALTER TABLE ADD COLUMN',
-  'ALTER TABLE RENAME TO',
-  'ALTER TABLE DROP COLUMN',
-  'ALTER COLUMN SET OPTIONS',
-  'ALTER COLUMN DROP NOT NULL',
-  'ALTER COLUMN SET DATA TYPE',
-  'ALTER VIEW SET OPTIONS',
-  'ALTER MATERIALIZED VIEW SET OPTIONS',
```
Comment (m-ahmadkhan) [inline src/languages/bigquery/bigquery.formatter.ts:67]: Same goes for these ALTER commands. They go like:

```
ALTER SCHEMA mydataset
SET OPTIONS (
  ...
)
```

File: test/bigquery.test.ts [hunk 2]
```diff
@@ -301,4 +250,681 @@ describe('BigQueryFormatter', () => {
...
+              b BOOL
+            >
+          )
+          PARTITION BY _PARTITIONDATE
+          OPTIONS(
+            expiration_timestamp=TIMESTAMP "2025-01-01 00:00:00 UTC",
+            partition_expiration_days=1
+          )`;
+
+        // need to fix the formatting of y STRUCT<a ARRAY<STRING>OPTIONS(description ="desc2"), b BOOL>
```
Comment (m-ahmadkhan) [inline test/bigquery.test.ts:307]: This PR https://github.com/sql-formatter-org/sql-formatter/pull/281 had fixed STRUCT formatting before, but there still are cases

Comment (m-ahmadkhan) [reply test/bigquery.test.ts:307]: Not an issue. Updating the tests.

File: test/bigquery.test.ts [hunk 3]
```diff
@@ -301,4 +250,681 @@ describe('BigQueryFormatter', () => {
...
+          RETURNS FLOAT64
+          AS (x * y);`;
+        const expected = dedent`
+          ${createFunction}
+            mydataset.myFunc (x FLOAT64, y FLOAT64) RETURNS FLOAT64 AS (x * y);`;
+        expect(format(input)).toBe(expected);
+        expect(format(input, { aliasAs: 'always' })).toBe(expected);
+      });
+
+      // it shouldn't format the js code
```
[the remark this answers is not in the document]
Comment (m-ahmadkhan) [reply test/bigquery.test.ts:474]: Not an issue. Updating tests.

File: test/bigquery.test.ts [hunk 4]
```diff
@@ -301,4 +250,677 @@ describe('BigQueryFormatter', () => {
...
+              ${dropSchema}
+                mydataset RESTRICT`,
+          },
+        ];
+
+        testSqls.forEach(testSql => expect(format(testSql.input)).toBe(testSql.expected));
+      });
+    });
+
+    const dropTables = ['DROP TABLE', 'DROP TABLE IF EXISTS'];
```
Comment (inferrinizzard) [inline test/bigquery.test.ts:857]: these lists can be declared as a map object, ie.
```js
const dropKeywordTypes = {
  dropTables: [...],
  // etc.
}
```

and you can collect them with the `flatKeywordLists` util function

**Comment by inferrinizzard** [comment]
Looks fairly comprehensive, thanks for the good work!

File: test/bigquery.test.ts [hunk 5]
```diff
@@ -301,4 +250,677 @@ describe('BigQueryFormatter', () => {
...
+      it(`Supports ${createFunction}`, () => {
+        const input = `
+          ${createFunction} mydataset.myFunc(x FLOAT64, y FLOAT64)
+          RETURNS FLOAT64
+          AS (x * y);`;
+        const expected = dedent`
+          ${createFunction}
+            mydataset.myFunc (x FLOAT64, y FLOAT64) RETURNS FLOAT64 AS (x * y);`;
+        expect(format(input)).toBe(expected);
+        expect(format(input, { aliasAs: 'always' })).toBe(expected);
```
Comment (nene) [inline test/bigquery.test.ts:471]: Huh, huh... finally figured out that this got fixed by the addition of `RETURNS` to list of keywords.

Should at least add a comment here mentioning that it's a regression test for issue #309

**Review by nene — COMMENTED** [review]
Hey. This looks great. Thanks for all the effort you've put into it. Especially appreciating all the tests with BigQuery code examples.

Wrote some suggestions. Mostly minor things about tests.

File: src/languages/bigquery/bigquery.formatter.ts [hunk 6]
```diff
@@ -33,57 +33,113 @@ const reservedCommands = [
   // DDL, https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language
   'SET SCHEMA', // added
```
Comment (m-ahmadkhan) [inline src/languages/bigquery/bigquery.formatter.ts:34]: Btw, I couldn't find SET SCHEMA anywhere in https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language.

File: test/bigquery.test.ts [hunk 7]
```diff
@@ -301,4 +250,677 @@ describe('BigQueryFormatter', () => {
...
+        const expected = dedent`
+          ${alterTable}
+            mydataset.mytable
+          SET DEFAULT COLLATE
+            'und:ci'`;
+        expect(format(input)).toBe(expected);
+      });
+
+      alterColumns.forEach(alterColumn => {
+        it(`Supports ${alterTable} - ${alterColumn} - SET OPTIONS`, () => {
```
[the remark this answers is not in the document]
Comment (m-ahmadkhan) [reply test/bigquery.test.ts]: You're right. Testing all the combinations doesn't make much sense. I've restructured the tests now.

File: src/languages/bigquery/bigquery.formatter.ts (hunk 6 above)
Comment (nene) [reply src/languages/bigquery/bigquery.formatter.ts:34]: Indeed, looks like it's here by mistake and can be removed.

**Review by nene — APPROVED** [review]

File: src/languages/bigquery/bigquery.formatter.ts (hunk 6 above)
Comment (nene) [reply src/languages/bigquery/bigquery.formatter.ts:34]: I dropped it by myself.

**Comment by nene** [comment]
Merged in and released as `9.0.0-beta2`

**Pull request merged**